blocks/bricks/__init__.py

```diff
@@ -158,7 +159,56 @@ def _push_initialization_config(self):
                     child.biases_init = self.biases_init
 
 
-class Linear(Initializable):
+class Feedforward(Brick):
+    """Declares an interface for bricks with one input and one output.
+
+    Many bricks have just one input and just one output (activations,
+    :class:`Linear`, :class:`MLP`). To make such bricks interchangable
+    in most contexts they should share an interface for configuring
+    their input and output dimensions. This brick provides such an
+    interface.
+
+    """
+    @abstractproperty
+    def input_dim():
+        """The input dimension of the brick."""
+        pass
+
+    @abstractproperty
+    def output_dim():
+        """The output dimension of the brick."""
+        pass
+
+
+def override_with_attribute(property_, attribute=None):
```

Comment:
> It gives an error when a class does not implement the property at all.

Right, but this is just what I wanted in fact. 

There two ways to declare  interfaces: in the code or in the docs. The disadvantage of the second one is that nothing is checked. I am trying to use the first one here to gently remind the user that he forgot to implement something which is required. Indeed it comes with a price: one can not just use an attribute with same name, one has to override the property.
To avoid boiler-plate code that does it, I implemented `override_with_attribute`.

So this is a typical dilemma of security vs. readability. I can switch to using java-like setters and getters, `get_input_dim`, etc.